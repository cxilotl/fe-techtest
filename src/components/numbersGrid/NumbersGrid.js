import React from 'react';
import PropTypes from 'prop-types';
import NumberBox from '../numberBox/NumberBox';
import cssStyles from './NumbersGrid.css';
import cssItemStyles from './NumbersGridItem.css';

const NumbersGridItem = ({ children }) => {
  return (
    <>
      <li className={ cssItemStyles.layout }>
        { children }
      </li>
    </>
  );
};

NumbersGridItem.propTypes = {
  children: PropTypes.node.isRequired,
};

const NumbersGrid = ({ maxNumToDisplay, selectedNum, highlightedNumbers, onNumberClickFn }) => {
  const onClickFn = (clickEvent) => {
    const selectedValue = parseInt(clickEvent.currentTarget.value, 10);
    onNumberClickFn(selectedValue);
  };

  const renderNumberList = (highestNumber, selectedValue, highlightedValues) => {
    let index = 1;
    const numberList = [];
    if (typeof highestNumber === "number" && highestNumber > 0) {
      for (index; index < highestNumber; index++) {
        const keyIndex = `number-id-${index}`;
        const isSelected = (selectedValue)? index === selectedValue : false;
        let isMultiple = false;
        if (!isSelected && highlightedValues.includes(index)) {
          isMultiple = true;
        }
        numberList.push(
          <NumbersGridItem key={ keyIndex }>
            <NumberBox numBoxLabel={ index } isSelected={ isSelected } isHighlighted={ isMultiple } onClickFn={ onClickFn } />
          </NumbersGridItem>
        );
      }
    }
    return numberList;
  };

  return (
    <>
      <ol className={ cssStyles.layout }>
        { renderNumberList(maxNumToDisplay, selectedNum, highlightedNumbers) }
      </ol>
    </>
  );
};

NumbersGrid.propTypes = {
  maxNumToDisplay: PropTypes.number,
  selectedNum: PropTypes.number,
  highlightedNumbers: PropTypes.arrayOf(PropTypes.number),
  onNumberClickFn: PropTypes.func
};

NumbersGrid.defaultProps = {
  maxNumToDisplay: 0,
  selectedNum: 0,
  highlightedNumbers: [],
  onNumberClickFn: () => {}
};

export default NumbersGrid;