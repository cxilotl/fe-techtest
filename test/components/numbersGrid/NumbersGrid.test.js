import React from 'react';
import renderer from 'react-test-renderer';
import NumbersGrid from '../../../src/components/numbersGrid/NumbersGrid';

describe('NumbersGrid', () => {
  it('snapshot', () => {
    const maxOfNumbers = 5;
    const component = renderer.create(<NumbersGrid maxNumToDisplay={maxOfNumbers} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});