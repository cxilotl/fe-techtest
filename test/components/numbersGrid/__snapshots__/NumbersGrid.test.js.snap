// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumbersGrid snapshot 1`] = `
<ol
  className="layout"
>
  <li
    className="layout"
  >
    <button
      className="layout"
      name="num-button-1"
      onClick={[Function]}
      type="button"
      value={1}
    >
      1
    </button>
  </li>
  <li
    className="layout"
  >
    <button
      className="layout"
      name="num-button-2"
      onClick={[Function]}
      type="button"
      value={2}
    >
      2
    </button>
  </li>
  <li
    className="layout"
  >
    <button
      className="layout"
      name="num-button-3"
      onClick={[Function]}
      type="button"
      value={3}
    >
      3
    </button>
  </li>
  <li
    className="layout"
  >
    <button
      className="layout"
      name="num-button-4"
      onClick={[Function]}
      type="button"
      value={4}
    >
      4
    </button>
  </li>
</ol>
`;
